front/src/routes/integration/all/mqtt/setup-page/actions.js

```diff
@@ -0,0 +1,67 @@
+import createActionsIntegration from '../../../../../actions/integration';
+import { RequestStatus } from '../../../../../utils/consts';
+
+const createActions = store => {
+  const integrationActions = createActionsIntegration(store);
+  const actions = {
+    async loadProps(state) {
+      let mqttURL;
+      let mqttUsername;
+      let mqttPassword;
+      try {
+        mqttURL = await state.httpClient.get('/api/v1/service/mqtt/variable/MQTT_URL');
+        mqttUsername = await state.httpClient.get('/api/v1/service/mqtt/variable/MQTT_USERNAME');
+        if (mqttUsername.value) {
+          mqttPassword = 'FAKE_PASSWORD';
+        }
+      } finally {
+        store.setState({
+          mqttURL: (mqttURL || {}).value,
+          mqttUsername: (mqttUsername || { value: '' }).value,
+          mqttPassword,
+          passwordChanges: false
+        });
+      }
+    },
+    updateConfigration(state, e) {
+      const data = {};
+      data[e.target.name] = e.target.value;
+      if (e.target.name === 'mqttPassword') {
+        data.passwordChanges = true;
+      }
+      store.setState(data);
+    },
+    async saveConfiguration(state) {
+      event.preventDefault();
+      store.setState({
+        connectMqttStatus: RequestStatus.Getting
+      });
+      try {
+        await state.httpClient.post('/api/v1/service/mqtt/variable/MQTT_URL', {
+          value: state.mqttURL
+        });
+        await state.httpClient.post('/api/v1/service/mqtt/variable/MQTT_USERNAME', {
+          value: state.mqttUsername
+        });
+        if (state.passwordChanges) {
```

Comment:
Why do you save everything all the time, and the password only if it changes? Just asking.

Reply:
Because I do not retrieve the real one from param table, so in the field, the value will be FAKE_PASSWORD by default, and in don't want to save it as is.

Reply:
Got it! The problem is that, if the user password is "FAKE_PASSWORD", it messes up everything right ?

We can maybe use a Symbol instead ( https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Symbol ) to avoid using a string const :)